lib/groonga-query-log/command/check-performance-regression.rb

```diff
@@ -0,0 +1,389 @@
+            end
+          end
+
+          parser.on("--input-old-query=PATH",
+                    "Use PATH as old input query log or PATH to old query log's directory.",
+                    "(#{@options[:input_old_query]})") do |path|
+            if File.directory?(path)
+              @options[:input_old_query] = Dir.glob("#{path}/*.log")
+            elsif File.exist?(path)
+              @options[:input_old_query] = [path]
+            else
+              raise OptionParser::InvalidOption.new("path <#{path}> doesn't exist")
+            end
+          end
+
+          parser.on("--input-new-query=PATH",
+                    "Use PATH as new input query log or PATH to new query log's directory",
+                    "(#{@options[:input_new_query]})") do |path|
+            if File.directory?(path)
+              @options[:input_new_query] = Dir.glob("#{path}/*.log")
+            elsif File.exist?(path)
+              @options[:input_new_query] = [path]
+            else
+              raise OptionParser::InvalidOption.new("path <#{path}> doesn't exist")
+            end
+          end
+
+          parser.on("--slow-operation-ratio=PERCENTAGE",
+                    Float,
+                    "Use PERCENTAGE% as threshold to detect slow operations.",
+                    "Example: --slow-operation-ratio=#{@options[:slow_operation_ratio]} means",
+                    "changed amount of operation time is #{@options[:slow_operation_ratio]}% or more.",
+                    "(#{@options[:slow_operation_ratio]})") do |ratio|
+            @options[:slow_operation_ratio] = ratio
+          end
+
+          parser.on("--slow-response-ratio=PERCENTAGE",
+                    Float,
+                    "Use PERCENTAGE% as threshold to detect slow responses.",
+                    "Example: --slow-response-ratio=#{@options[:slow_response_ratio]} means",
+                    "changed amount of response time is #{@options[:slow_response_ratio]}% or more.",
+                    "(#{@options[:slow_response_ratio]})") do |ratio|
+            @options[:slow_response_ratio] = ratio
+          end
+
+          parser.on("--slow-operation-threshold=THRESHOLD",
+                    Float,
+                    "Use THRESHOLD seconds to detect slow operations.",
+                    "(#{@options[:slow_operation_threshold]})") do |threshold|
+            @options[:slow_operation_threshold] = threshold
+          end
+
+          parser.on("--slow-response-threshold=THRESHOLD",
+                    Float,
+                    "Use THRESHOLD seconds to detect slow responses.",
+                    "(#{@options[:slow_response_threshold]})") do |threshold|
+            @options[:slow_response_threshold] = threshold
+          end
+        end
+      end
+
+      def group_statistics(old_statistics, new_statistics)
+        old_queries = {}
+        new_queries = {}
+        old_statistics.count.times do |i|
```

Comment:
Can we use `old_statistics.zip(new_statistics) do |old_statistic, new_statistic|` here?

Reply:
fixed.